lib/types/local-time.js

```diff
@@ -27,257 +30,236 @@ const nanoSecInMillis = Long.fromNumber(1000000);
  * @private
  * */
 const millisInDay = 86400000;
+
 /**
+ * A time without a time-zone in the ISO-8601 calendar system, such as 10:30:05.
  *
- * Creates a new instance of LocalTime.
- * @class
- * @classdesc A time without a time-zone in the ISO-8601 calendar system, such as 10:30:05.
- * <p>
- *   LocalTime is an immutable date-time object that represents a time, often viewed as hour-minute-second. Time is represented to nanosecond precision. For example, the value "13:45.30.123456789" can be stored in a LocalTime.
- * </p>
- * @param {Long} totalNanoseconds Total nanoseconds since midnight.
- * @constructor
+ * LocalTime is an immutable date-time object that represents a time, often viewed as hour-minute-second. Time is represented to nanosecond precision. For example, the value "13:45.30.123456789" can be stored in a LocalTime.
  */
-function LocalTime(totalNanoseconds) {
-    if (!(totalNanoseconds instanceof Long)) {
-        throw new Error("You must specify a Long value as totalNanoseconds");
-    }
-    if (
-        totalNanoseconds.lessThan(Long.ZERO) ||
-        totalNanoseconds.greaterThan(maxNanos)
-    ) {
-        throw new Error("Total nanoseconds out of range");
+class LocalTime {
+    #internal;
+
+    /**
+     * Creates a new instance of LocalTime.
+     * @param {Long} totalNanoseconds Total nanoseconds since midnight.
+     */
+    constructor(totalNanoseconds) {
+        if (!(totalNanoseconds instanceof Long)) {
+            throw new TypeError(
+                "You must specify a Long value as totalNanoseconds",
+            );
+        }
+        if (
+            totalNanoseconds.lessThan(Long.ZERO) ||
+            totalNanoseconds.greaterThan(maxNanos)
+        ) {
+            throw new TypeError("Total nanoseconds out of range");
+        }
+
+        this.#internal = rust.LocalTimeWrapper.new(
+            longToBigint(totalNanoseconds),
+        );
+
+        this.value = totalNanoseconds;
     }
-    this.value = totalNanoseconds;
 
     /**
-     * Gets the hour component of the time represented by the current instance, a number from 0 to 23.
-     * @type Number
+     * Gets the nanoseconds, a number from 0 to 999 999 999.
+     * @readonly
+     * @type {Number}
      */
-    this.hour = this._getParts()[0];
+    get nanosecond() {
+        return this.#internal.nanosecond;
+    }
+
+    set nanosecond(newValue) {
+        throw new SyntaxError("Nanosecond length is read-only");
```

Comment:
Make error message consisten with [other types](https://github.com/scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver/blob/c4710a932841e38bc7bd9ea65b644554ab45608b/lib/types/uuid.js#L45-L47):
``{Type} {field} is read-only``